### PR TITLE
Add more tests for TreeOfLosers

### PR DESCRIPTION
**Pull request by xiaoxmeng**
Add more tests for TreeOfLosers to catch the previous bug in TreeOfLosers
caught in aggregation spilling test.

### CODE REVIEW

File: velox/exec/tests/TreeOfLosersTest.cpp [hunk 1]
```diff
@@ -100,3 +100,144 @@ TEST_F(TreeOfLosersTest, singleWithEquals) {
...
+      stream->pop();
+    }
+  }
+}
+
+TEST_F(TreeOfLosersTest, allSorted) {
+  const int kNumsPerStream = 40;
+  const int kNumStreams = 20;
+  const uint32_t kStartValue = 10;
+  for (bool testNextEqual : {/*false, */ true}) {
```
Comment (Yuhta) [inline velox/exec/tests/TreeOfLosersTest.cpp:145]: We can test `next()` as well here

**Review by Yuhta — APPROVED** [review]

**Comment by pedroerp** [comment]
Thanks @xiaoxmeng ! It would be super nice to have micro-benchmarks for tree of losers under velox/benchmarks/basic as well :)

**Comment by xiaoxmeng** [comment]
> Thanks @xiaoxmeng ! It would be super nice to have micro-benchmarks for tree of losers under velox/benchmarks/basic as well :)

Will do. Thanks for suggestion!

File: velox/exec/tests/TreeOfLosersTest.cpp (hunk 1 above)
Comment (xiaoxmeng) [reply velox/exec/tests/TreeOfLosersTest.cpp:145]: Yeah, I commented out while debugging the test code. Thanks for pointing out!

**Pull request closed without merging**